core/client/fs/src/main/java/alluxio/client/file/FileInStream.java

```diff
@@ -312,42 +319,49 @@ private void closeBlockInStream(BlockInStream stream) throws IOException {
       if (blockSource == BlockInStream.BlockInStreamSource.LOCAL) {
         return;
       }
+      triggerAsyncCaching(stream);
```

Comment:
hmm, this is redundant for pread

Reply:
The bigger concern I have is that we will call `triggerAsyncCaching` repeatedly for the same positioned read stream. For example if we issue 1000 preads, we will also issue 1000 calls for async caching. Could we write some logic associated with the cached pread stream to ensure we only call trigger async caching once? That would solve this problem as well.

Reply:
changed the logic to: send async cache request if blockId != <block Id for which last request was sent>